shimming-toolbox/shimmingtoolbox/cli/b0shim.py

```diff
@@ -322,21 +322,33 @@ def dynamic(fname_fmap, fname_anat, fname_mask_anat, method, opt_criteria, slice
                 logger.debug("Converting Siemens scanner coil from Shim CS (LAI) to Gradient CS")
                 # First convert to RAS
                 orders = tuple([order for order in scanner_coil_order if order != 0])
-                for i_shim in range(coefs.shape[0]):
-                    # Convert coefficient
-                    coefs_coil[i_shim, 1:] = shim_to_phys_cs(coefs_coil[i_shim, 1:], manufacturer, orders)
 
-                # Convert coef of 1st order sph harmonics to Gradient coord system
-                coefs_freq, coefs_phase, coefs_slice = phys_to_gradient_cs(coefs_coil[:, 1],
-                                                                           coefs_coil[:, 2],
-                                                                           coefs_coil[:, 3], fname_anat)
+                if 0 in scanner_coil_order:
+                    for i_shim in range(coefs.shape[0]):
+                        # Convert coefficient
+                        coefs_coil[i_shim, 1:] = shim_to_phys_cs(coefs_coil[i_shim, 1:], manufacturer, orders)
+                    # Convert coef of 1st order sph harmonics to Gradient coord system
+                    coefs_freq, coefs_phase, coefs_slice = phys_to_gradient_cs(coefs_coil[:, 1],
```

Comment:
This seems to keep the same issue if I select order 0 and 2

Reply:
I would add a specification in the error message for supported orders

Reply:
line 259

Reply:
Done in [48e17b5](https://github.com/shimming-toolbox/shimming-toolbox/pull/560/commits/48e17b58778e642d35d2201835c259708abc99e9)